ToDo.Core/ViewModels/ToDoViewModel.cs

```diff
@@ -40,6 +44,23 @@ public ToDoViewModel(IToDoService todoService, IMvxNavigationService navigationS
                 async (swipedItem) => await RemoveTodo((Models.ToDo)swipedItem));
             EditTodoItemCommand = new Command(
                 async (swipedItem) => await EditTodo((Models.ToDo)swipedItem));
+            SearchTextChangedCommand = new Command(
+                async (query) => await SearchedTextChanged((string)query));
+
+            _subscription = this.WhenAnyValue(x => x.SearchedString)
+                .Throttle(TimeSpan.FromMilliseconds(500))
+                .Where(x => !string.IsNullOrWhiteSpace(x))
+                .ObserveOn(RxApp.MainThreadScheduler)
+                .SelectMany(x =>
+                {
+                    return SearchTodos(x).ContinueWith((a) => Task.FromResult(Unit.Default));
+                })
+                .Subscribe();
+        }
+
+        ~ToDoViewModel()
```

Comment:
It should never really be necessary to create a destructor in C#
I guess you have this habit from C++?

Reply:
Or maybe from C. 🙈 But I guess I should call _subscription.Dispose() somewhere, right? So if not in the destructor, then where?

Reply:
In this case, you don't need to dispose the subscription, because you only use `this.WhenAnyValue`. There are no references to other objects that have to be cleared with a `.Dispose()`